core/src/main/java/org/apache/felix/ipojo/handlers/dependency/Dependency.java

```diff
@@ -661,13 +673,15 @@ public Object getService() {
         Usage usage = (Usage) m_usage.get();
```

Comment:
Here there is no null check on m_usage... it is a bit sad to have an inconsistency...

Can it really be null on lines 595 & 611 ?

Reply:
Honestly, hard to say

In some cases it is checked, like in onEntry()/onExit()
Since the initialization is done with
```
        if (field != null) {
            m_usage = new ServiceUsage();
        } else {
            m_usage = null;
        }```
I'm guessing it has to do with the code paths depending on if a field was provided or not
I didn't have enough knowledge and didn't want to risk it so I just did the check :)